cpp/src/arrow/table.cc

```diff
@@ -601,6 +602,19 @@ TableBatchReader::TableBatchReader(const Table& table)
   }
 }
 
+TableBatchReader::TableBatchReader(std::shared_ptr<Table> table)
+    : table_(*table.get()),
+      owned_table_(std::move(table)),
```

Comment:
This should probably be declared before `table_` since `table_` is `*owned_table_`. It doesn't matter much now, but if another member that takes `Table &` were to be introduced here, it would have a `Table &` pointing to freed memory when everything is destroyed.

Reply:
Thanks for pointing that out! Made the change.